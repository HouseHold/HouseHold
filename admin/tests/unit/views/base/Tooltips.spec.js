import Vue from 'vue'
import { shallowMount } from '@vue/test-utils';
import CoreuiVue from '@coreui/vue'
import Tooltips from '@/views/base/Tooltips'

Vue.use(CoreuiVue)

describe('Tooltips.vue', () => {
  it('has a name', () => {
    expect(Tooltips.name).toBe('Tooltips')
  })
  it('has a created hook', () => {
    expect(typeof Tooltips.data).toMatch('function')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Tooltips)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Tooltips', () => {
    const wrapper = shallowMount(Tooltips)
    expect(wrapper.is(Tooltips)).toBe(true)
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Tooltips)
    expect(wrapper.element).toMatchSnapshot()
  })
})
