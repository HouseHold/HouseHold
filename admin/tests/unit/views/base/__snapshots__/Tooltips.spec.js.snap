// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltips.vue renders correctly 1`] = `
<div>
  <ccard-stub>
    <ccardheader-stub>
      <cicon-stub
        name="cil-justify-center"
      />
       
      <strong>
         Bootstrap Tooltips 
      </strong>
       
      <small>
        <code>
          v-c-tooltip
        </code>
         directive
      </small>
       
      <div
        class="card-header-actions"
      >
        <a
          class="card-header-action"
          href="https://coreui.io/vue/docs/directives/tooltip"
          rel="noreferrer noopener"
          target="_blank"
        >
          <small
            class="text-muted"
          >
            docs
          </small>
        </a>
      </div>
    </ccardheader-stub>
     
    <ccardbody-stub>
      <crow-stub
        gutters="true"
        tag="div"
      >
        <ccol-stub
          col="6"
          tag="div"
        >
          <div
            class="text-center my-3"
          >
            <cbutton-stub
              activeclass="active"
              color="secondary"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              Hover Me
            
            </cbutton-stub>
          </div>
        </ccol-stub>
         
        <ccol-stub
          col="6"
          tag="div"
        >
          <div
            class="text-center my-3"
          >
            <cbutton-stub
              activeclass="active"
              color="secondary"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              Hover me
            
            </cbutton-stub>
          </div>
        </ccol-stub>
      </crow-stub>
    </ccardbody-stub>
  </ccard-stub>
   
  <ccard-stub>
    <ccardheader-stub>
      <cicon-stub
        name="cil-justify-center"
      />
       
      <strong>
         Tooltips 
      </strong>
       
      <small>
        placement
      </small>
    </ccardheader-stub>
     
    <ccardbody-stub>
      <div
        class="my-3"
      >
        <crow-stub
          gutters="true"
          tag="div"
        >
          <ccol-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <cbutton-stub
              activeclass="active"
              color="primary"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              top-start
            
            </cbutton-stub>
          </ccol-stub>
          <ccol-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <cbutton-stub
              activeclass="active"
              color="primary"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              top
            
            </cbutton-stub>
          </ccol-stub>
          <ccol-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <cbutton-stub
              activeclass="active"
              color="primary"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              top-end
            
            </cbutton-stub>
          </ccol-stub>
          <ccol-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <cbutton-stub
              activeclass="active"
              color="primary"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              bottom-start
            
            </cbutton-stub>
          </ccol-stub>
          <ccol-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <cbutton-stub
              activeclass="active"
              color="primary"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              bottom
            
            </cbutton-stub>
          </ccol-stub>
          <ccol-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <cbutton-stub
              activeclass="active"
              color="primary"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              bottom-end
            
            </cbutton-stub>
          </ccol-stub>
          <ccol-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <cbutton-stub
              activeclass="active"
              color="primary"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              right-start
            
            </cbutton-stub>
          </ccol-stub>
          <ccol-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <cbutton-stub
              activeclass="active"
              color="primary"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              right
            
            </cbutton-stub>
          </ccol-stub>
          <ccol-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <cbutton-stub
              activeclass="active"
              color="primary"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              right-end
            
            </cbutton-stub>
          </ccol-stub>
          <ccol-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <cbutton-stub
              activeclass="active"
              color="primary"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              left-start
            
            </cbutton-stub>
          </ccol-stub>
          <ccol-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <cbutton-stub
              activeclass="active"
              color="primary"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              left
            
            </cbutton-stub>
          </ccol-stub>
          <ccol-stub
            class="py-4 text-center"
            md="4"
            tag="div"
          >
            <cbutton-stub
              activeclass="active"
              color="primary"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
              type="button"
            >
              
              left-end
            
            </cbutton-stub>
          </ccol-stub>
        </crow-stub>
      </div>
    </ccardbody-stub>
  </ccard-stub>
</div>
`;
